server/src/main/java/com/cloud/network/router/CommandSetupHelper.java

```diff
@@ -840,10 +844,18 @@ public int compare(final PublicIpAddress o1, final PublicIpAddress o2) {
 
             for (final PublicIpAddress ipAddr : ipAddrList) {
                 final boolean add = ipAddr.getState() == IpAddress.State.Releasing ? false : true;
+                String vlanTag = ipAddr.getVlanTag();
+                String key = null;
+                if (Objects.isNull(vlanTag)) {
+                    key = "nsx-" + ipAddr.getAddress().addr();
```

Comment:
Why the nsx prefix is needed? Is there any other case in which the vlan tag may be empty but is not an NSX VPC?

Reply:
No - as of now, I dont think so..